### ISSUE TITLE
Are there any difference of autograd between these two ways to get the gradient of FFT?

### ISSUE THREAD

**Issue by taoxuewen** [pytorch/pytorch#15377]
## ❓ Questions and Help
In https://pytorch.org/docs/stable/notes/autograd.html, I got that if I want to compute the gradient of fft, only need to code:
import torch
a=torch.randn([1,1024],requires_grad=True)
b=torch.rfft(a,1)
c=torch.sum(b)
c.backward()
However, https://pytorch.org/tutorials/advanced/numpy_extensions_tutorial.html provided a method which combined numpy.fft and pytorch to do it. So I want to ask that are there any difference between them?
Thank you!

### Please note that this issue tracker is not a help form and this issue will be closed.

We have a set of [listed resources available on the website](https://pytorch.org/resources). Our primary means of support is our discussion forum:

- [Discussion Forum](https://discuss.pytorch.org/)

**Comment by SsnL** [comment]
pytorch native fft works on cuda, and should be preferred as we have extensive tests on it. the tutorial is just an example of how to write extensions.
